katz/src/main/kotlin/katz/arrow/FunctionK.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2017 The Katz Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package katz
+
+interface FunctionK<F, G> {
+
+    /**
+     * Applies this functor transformation from `F` to `G`
+     */
+    operator fun <A> invoke(fa: HK<F, A>): HK<G, A>
+
+    fun <H> or(h: FunctionK<H, G>): FunctionK<CoproductFG<F, H>, G> =
```

Comment:
I believe this will work better as an extension function, if there are no impediments. That way the interface is kept clean from default implementations.